zilliqa/src/api/types/zil.rs

```diff
@@ -31,14 +31,16 @@ pub struct TxBlock {
     pub body: TxBlockBody,
 }
 
-impl From<&Block> for TxBlock {
-    fn from(block: &Block) -> Self {
+impl TxBlock {
+    pub fn new(block: &Block, proposer: Address) -> Self {
         // TODO(#79): Lots of these fields are empty/zero and shouldn't be.
+        let mut scalar = [0; 32];
+        scalar[31] = 1;
         TxBlock {
             header: TxBlockHeader {
-                version: 0,
-                gas_limit: transaction::ScillaGas(0),
-                gas_used: 0,
+                version: 2,                                    // ZQ2
```

Comment:
If we're doing this - Should we take care to keep the `version` for converted ZQ1 blocks as `0`? Or alternatively just don't expose the difference between ZQ1 and ZQ2 blocks in this field (it is implied by block height anyway).

Reply:
This might be another case of something we don't want to bring over from ZQ1 - my reasoning was that there was a version field there, so if we also return a version number in ZQ2, it's less likely to break anyone's code that depends on it.

Overall it's a question of how much backwards compatibility we're targetting